Source/Core/Core/PowerPC/JitArm64/JitArm64_SystemRegisters.cpp

```diff
@@ -623,25 +602,55 @@ void JitArm64::crXXX(UGeckoInstruction inst)
   }
 
   ORR(XB, XB, 32, 0, true);  // XB | 1<<32
-  STR(INDEX_UNSIGNED, XB, PPC_REG, PPCSTATE_OFF(cr_val[field]));
 
   gpr.Unlock(WA);
-  gpr.Unlock(WB);
 }
 
 void JitArm64::mfcr(UGeckoInstruction inst)
 {
   INSTRUCTION_START
   JITDISABLE(bJITSystemRegistersOff);
 
-  gpr.Lock(W0, W1, W2, W30);
-  BL(GetAsmRoutines()->mfcr);
-  gpr.Unlock(W1, W2, W30);
-
   gpr.BindToRegister(inst.RD, false);
-  MOV(gpr.R(inst.RD), W0);
+  ARM64Reg WA = gpr.R(inst.RD);
+  ARM64Reg WB = gpr.GetReg();
+  ARM64Reg WC = gpr.GetReg();
+  ARM64Reg XA = EncodeRegTo64(WA);
+  ARM64Reg XB = EncodeRegTo64(WB);
+  ARM64Reg XC = EncodeRegTo64(WC);
+
+  for (int i = 0; i < 8; i++)
+  {
+    MOV(XB, gpr.CR(i));
 
-  gpr.Unlock(W0);
+    // SO
+    if (i == 0)
+    {
+      UBFX(XA, XB, 61, 1);
+    }
+    else
+    {
+      ORR(WA, WZR, WA, ArithOption(WA, ST_LSL, 4));
+      UBFX(XC, XB, 61, 1);
+      ORR(XA, XA, XC);
```

Comment:
As the upper bits of WA are zero, this is the same as:
UBFX(XC, XB, 61, 1);
ORR(WA, XC, XA, ArithOption(XA, ST_LSL, 4));